plugins/modules/database/misc/odbc.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, John Westcott <john.westcott.iv@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: odbc
+author: "John Westcott IV (@john-westcott-iv)"
+version_added: "2.8"
+short_description: Execute SQL via ODBC
+description:
+    - Read/Write info via ODBC
+options:
+    dsn:
+      description:
+        - The connection string passed into ODBC
+      required: True
+      type: str
+    query:
+      description:
+        - The SQL query to perform
+      required: True
+      type: str
+    params:
+      description:
+        - Parameters to pass to the SQL squery
+      type: list
+      elements: str
+
+requirements:
+  - "python >= 2.6"
+  - "pyodbc"
+'''
+
+EXAMPLES = '''
+- name: Set some values in the test db
+  odbc:
+    dsn: "DRIVER={ODBC Driver 13 for SQL Server};Server=db.ansible.com;Database=my_db;UID=admin;PWD=password;"
+    query: "Select * from table_a where column1 = ?"
+    params:
+      - "value1"
+'''
+
+RETURN = '''
+results:
+    description: List of dicts containing selected rows, likley empty for DDL statements.
+    returned: success
+    type: list
+description:
+    description: "List of dicts about the columns selected from the cursors. See https://github.com/mkleehammer/pyodbc/wiki/Cursor"
+    returned: success
+    type: list
+row_count:
+    description: "The number of rows selected or modified according to the cursor. See https://github.com/mkleehammer/pyodbc/wiki/Cursor"
+    returned: success
+    type: str
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+HAS_PYODBC = None
+try:
+    import pyodbc
+    HAS_PYODBC = True
+except ImportError as e:
+    HAS_PYODBC = False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            dsn=dict(type='str', required=True, no_log=True),
+            query=dict(type='str', required=True),
+            params=dict(type='list', elements='str'),
+        ),
+    )
+
+    dsn = module.params.get('dsn')
+    query = module.params.get('query')
+    params = module.params.get('params')
+
+    if not HAS_PYODBC:
+        module.fail_json(msg='pyodbc python module must be installed to use this module')
```

Comment:
In ansible/ansible's module_utils there's a function which formats a nicer message (which includes information on the Python version and the machine this is run on).